src/main/java/io/netty/incubator/codec/quic/QuicheQuicCodec.java

```diff
@@ -179,6 +186,34 @@ public final void channelWritabilityChanged(ChannelHandlerContext ctx) {
         ctx.fireChannelWritabilityChanged();
     }
 
+    @Override
+    public final void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)  {
+        int size = estimatorHandle.size(msg);
+        if (size > 0) {
+            pendingBytes += size;
+        }
+        try {
+            ctx.write(msg, promise);
+        } finally {
+            // If the number of bytes pending exceeds the max batch size we should force a flush() and so ensure
+            // these are delivered in a timely manner and also make room in the outboundbuffer again that belongs
+            // to the underlying channel.
+            if (pendingBytes > maxBytesBeforeFlush) {
+                flushNow(ctx);
+            }
+        }
+    }
+
+    @Override
+    public final void flush(ChannelHandlerContext ctx) {
+        flushNow(ctx);
+    }
+
+    private void flushNow(ChannelHandlerContext ctx) {
+        pendingBytes = 0;
+        ctx.flush();
+    }
```

Comment:
Why separate `flush` and `flushNow` methods?

Reply:
I just liked it a bit more :) It not really matters.

Reply:
also I just noticed we can "suppress" the original flush if pendingBytes == 0